std/kit/map.kit

```diff
@@ -6,6 +6,12 @@ trait Hashable {
     function hash(): Int;
 }
```

Comment:
Since this is shared now, maybe it should go in a separate "kit.hash" module.

Reply:
That’s what I suggested a while back, move the traits and their implementations for standard types to a new file. I can do that :)